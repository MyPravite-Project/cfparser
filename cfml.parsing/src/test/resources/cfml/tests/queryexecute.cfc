        var result = QueryExecute("
            SELECT info_card_id, document_num, revision_nm, title_nm
            FROM tdc_doc_infocard
            WHERE info_card_id = :sourceTemplateNum
            ORDER BY UPPER(document_num) ASC, revision_seq DESC
        ", { sourceTemplateNum = { value="#sourceTemplateNum#", cfsqltype="cf_sql_varchar" }});