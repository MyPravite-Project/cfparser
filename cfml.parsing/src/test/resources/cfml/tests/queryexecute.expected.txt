/*===TOKENS===*/
VAR                      <var>
IDENTIFIER               <result>
'='                      <=>
IDENTIFIER               <QueryExecute>
'('                      <(>
STRING_LITERAL           <"
            SELECT info_card_id, document_num, revision_nm, title_nm
            FROM tdc_doc_infocard
            WHERE info_card_id = :sourceTemplateNum
            ORDER BY UPPER(document_num) ASC, revision_seq DESC
        ">
','                      <,>
'{'                      <{>
IDENTIFIER               <sourceTemplateNum>
'='                      <=>
'{'                      <{>
IDENTIFIER               <value>
'='                      <=>
STRING_LITERAL           <"#sourceTemplateNum#">
','                      <,>
IDENTIFIER               <cfsqltype>
'='                      <=>
STRING_LITERAL           <"cf_sql_varchar">
'}'                      <}>
'}'                      <}>
')'                      <)>
';'                      <;>
/*===TREE===*/
(scriptBlock
  (element
    (statement
      (localAssignmentExpression
        var
        (identifier result)
        =
        (startExpression
          (compareExpression
            (baseExpression
              (unaryExpression
                (memberExpression
                  (functionCall
                    (identifierOrReservedWord (identifier QueryExecute))
                    (
                    (argumentList
                      (argument
                        (startExpression
                          (compareExpression
                            (baseExpression
                              (unaryExpression
                                (primaryExpression
                                  (literalExpression
                                    "\r\n            SELECT info_card_id, document_num, revision_nm, title_nm\r\n            FROM tdc_doc_infocard\r\n            WHERE info_card_id = :sourceTemplateNum\r\n            ORDER BY UPPER(document_num) ASC, revision_seq DESC\r\n        "
                                  )
                                )
                              )
                            )
                          )
                        )
                      )
                      ,
                      (argument
                        (startExpression
                          (compareExpression
                            (baseExpression
                              (unaryExpression
                                (primaryExpression
                                  (implicitStruct
                                    {
                                    (implicitStructElements
                                      (implicitStructExpression
                                        (implicitStructKeyExpression
                                          (multipartIdentifier (identifier sourceTemplateNum))
                                        )
                                        =
                                        (baseExpression
                                          (unaryExpression
                                            (primaryExpression
                                              (implicitStruct
                                                {
                                                (implicitStructElements
                                                  (implicitStructExpression
                                                    (implicitStructKeyExpression (multipartIdentifier (identifier value)))
                                                    =
                                                    (baseExpression
                                                      (unaryExpression (primaryExpression (literalExpression "#sourceTemplateNum#")))
                                                    )
                                                  )
                                                  ,
                                                  (implicitStructExpression
                                                    (implicitStructKeyExpression (multipartIdentifier (identifier cfsqltype)))
                                                    =
                                                    (baseExpression
                                                      (unaryExpression (primaryExpression (literalExpression "cf_sql_varchar")))
                                                    )
                                                  )
                                                )
                                                }
                                              )
                                            )
                                          )
                                        )
                                      )
                                    )
                                    }
                                  )
                                )
                              )
                            )
                          )
                        )
                      )
                    )
                    )
                  )
                )
              )
            )
          )
        )
      )
      ;
    )
  )
  (endOfScriptBlock <EOF>)
)
/*======*/